integrated_channels/sap_success_factors/models.py

```diff
@@ -143,3 +155,74 @@ def __repr__(self):
         Return uniquely identifying string representation.
         """
         return self.__str__()
+
+
+class SAPSuccessFactorsLearnerCompletionPayload(object):
+    """
+    Stores the learner completion payload to be sent to SAP SuccessFactors.
+
+    Serializes to the format required by SAP SuccessFactors.
+    """
+    empty_value = ''
+    epoch = datetime(1970, 1, 1, tzinfo=timezone.utc)
+
+    def __init__(self, **kwargs):
+        """
+        Store all keys and values provided as data for the instance.
+        """
+        self._data = kwargs
+
+    def __str__(self):
+        """
+        Return a JSON-serialized representation of the given data.
+
+        Sort the keys so the result is consistent and testable.
+        """
+        return json.dumps(self._data, sort_keys=True)
+
+    @classmethod
+    def create(cls, enterprise_enrollment, certificate):
+        """
+        Create and return a new SAPSuccessFactorsLearnerCompletionPayload object.
+
+        Initializing the SAP SuccessFactors named keys and values from the provided EnterpriseCourseEnrollment and
+        GeneratedCertificate objects.
+        """
+        # TODO: ENT-164 will provide a way to get the SAP User ID for this user from the SAP SSO backend.
+        user_id = str(enterprise_enrollment.enterprise_customer_user.user_id)
+        course_id = enterprise_enrollment.course_id
+
+        # Get provider_id from global configuration settings
+        global_config = SAPSuccessFactorsGlobalConfiguration.current()
+        provider_id = global_config.provider_id
+
+        if certificate is None:
+            completed_timestamp = None
+            course_completed = "false"
+            grade = cls.empty_value
+        else:
+            completed_timestamp = int((certificate.created_date - cls.epoch).total_seconds())
+            course_completed = "true"
+            grade = certificate.grade
+
+        # Get the price and currency from the enrollment seat
+        price = cls.empty_value
+        currency = cls.empty_value
+        credit_hours = cls.empty_value
+
+        return SAPSuccessFactorsLearnerCompletionPayload(
+            edxUserID=user_id,
```

Comment:
Yep, sorry about that.  Fixed in 37299d0 - see [`_payload_data()`](https://github.com/edx/edx-enterprise/pull/59/files#diff-b2d1ad7443d2344126a60d0213e2d192R174)